lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -365,7 +365,7 @@ def get_d3_section_grade_distrib(course_id, section):
                         c_problem += 1
                         problem_set.append(library_problem.location)
                         problem_info[library_problem.location] = {
-                            'id': library_problem.location.to_deprecated_string(),
+                            'id': text_type(library_problem.location),
```

Comment:
We're no longer using the `to_deprecated_string()` methods.

In Python 3, this will be `str(key)`.
In Python 2, this is `unicode(key)`

During the py2/3 transition, we'll use `six` for compatibility:
- `text_type(key)`

This will use the appropriate string method, depending on Python version.

But ultimately, after the py3 switchover, we'll switch these all to `unicode(key)`.